### ISSUE TITLE
Adding dates to the postpone function

### ISSUE THREAD

**Issue by Oroumov** [benrhughes/todotxt.net#361]
Hello Ben (and other contributors)...
...and first of all congratulations, and thanks, for the todotxt.net app ! This is a great productivity app.
I've known the Todotxt format and concept for years since Gina talked about it on LifeHacker, switched to it 9 months ago, and have been using your app since then.... it's simply awesome (and awesomely simple, which is what I need)
After a few months, there are a few features that I'd like to suggest that would make my (everyone's ?) daily work so much greater. 
(Also, I apologise if these features have been suggested in the 300+ topics below... I tried to search as best as I could)
I'll split this post for separate feature follow-up.
Thanks again so much for your work !

Feature : 
* Adding dates to the postpone function (P)

Description :
* The P feature currently supports a mix of a relative value [number of days, like "17"], and absolute values [a week day like "Tuesday", or a few of the keywords like "today", "tomorrow"], but if you want to postpone a date to Monday in 2 weeks, and you happen to know that date... then you can't do it except using, for this one time only, the Set (S) feature.. which I always think of after hitting P :) Extending further from this, I think that the S and P features could almost be combined, as setting or postponing to a fixed date is exactly the same, almost eliminating the need for the S feature... what do you think ?

Thanks,
Nicolas.

**Comment by benrhughes** [comment]
Hi,
As you outlined, Set and Postpone are different ways of doing the same
thing (setting the due date). I'm not sure how we'd combine them into one
sensible UI that would be easier to use than just using S or P depending on
if you want to set a relative or absolute date. But maybe I'm
misunderstanding the use case?

Cheers,

Ben

**Comment by Oroumov** [comment]
Hi Ben, 

Thanks for taking time to reply ! :)
I think I mixed two questions at once. Let me try to clarify.
I'll try to give an example. 

Very often, I review a task, think it has to be postponed as the current due date can't be met, intuitively type P, then for example we are Monday and I want to postpone to **next** Wednesday, so now I have to think "oh wait in fact it's more than exactly 7 days so I'm not going to be able to use P and I have to use S instead", so I then have to exit P (Esc), type S instead, type Tab once to move the focus from the "OK" button to the date field, type the new date "15/08/2018" (which by the way is different from 2018-08-15, but I'm guessing it's due to my European settings), then type Tab again to get back to the OK field (Enter won't work on the date field), and finally type Enter.

So my first idea was : as P already accepts heterogeneous inputs ("17", "wed", "today"), I was wondering if it could accept also absolute values like "2018-08-15", which would be the direct translation of "postpone this activity to the 15th of august", and would eliminate the need for the complex gymnastics above with the S function.

And the second idea that I tried to describe at the same time was the following : to me, set and postpone have ultimately the same goal which is to define a due time (or a new due time) on a task. It's conceptually almost the same to say "I create this task now due tomorrow" (set) or "I create this task now and postpone its completion to tomorrow" (could be set or postpone) or "I update this task which already had another due date and postpone its completion to tomorrow" (postpone). Ultimately the edited task is due tomorrow and that's only what matters.
That's where I jumped to conclusions and suggested out loud that this could become a single function.
Assuming the first idea above got your blessing, now the postpone function does all what S did; and now, maybe it no longer needs to be called Postpone but could be simply called "Edit due date" or another generic name like that and the user guide (or the tooltip) would be here to tell you that it supports not only yyyy-mm-dd (or a choice in the date picker), but also relative values such as "tomorrow" or "5".

Sorry for this long block, hopefully it will be clearer now and maybe I'll even convince you ! :D
Thanks for reading !

Cheers, 
Nicolas.
